pilot/pkg/networking/plugin/health/health.go

```diff
@@ -61,7 +61,9 @@ func buildHealthCheckFilters(filterChain *plugin.FilterChain, probes model.Probe
 		// Check that the probe matches the listener port. If not, then the probe will be handled
 		// as a management port and not traced. If the port does match, then we need to add a
 		// health check filter for the probe path, to ensure that health checks are not traced.
-		if probe.Port.Port == endpoint.Port {
+		// If no probe port is defined, then port has not specifically been defined, so assume filter
+		// needs to be applied.
+		if probe.Port == nil || probe.Port.Port == endpoint.Port {
```

Comment:
If port is nil how will kube pick the port ?

Reply:
As I understand it, if the container has multiple ports then prometheus will try to scrape from each of them unless the `prometheus.io/port` annotation is defined. So to cover this situation, this logic would result in a health check filter being created for each port.